### ISSUE TITLE
CommandSpec injected into Mixins doesn't have a CommandLine

### ISSUE THREAD

**Issue by zegelin** [remkop/picocli#439]
When a `CommandSpec` is injected into a `@Mixin` class, the `CommandSpec.commandLine()` method returns null.

This prevents custom setter methods from throwing `ParameterExceptions`, since the exception constructor requires a not-null `CommandLine` object.

```
    static class ExampleMixin {
        @Spec CommandSpec spec;

        @Option(names = "--trex")
        void setTRexFences(final String value) {
            throw new CommandLine.ParameterException(spec.commandLine(), "Didn't say the magic word.");
        }
    }

    static class ExampleCommand implements Callable<Void> {
        @Spec CommandSpec spec;

        @Option(names = "--raptor")
        void setRaptorFences(final String value) {
            throw new CommandLine.ParameterException(spec.commandLine(), "Didn't say the magic word.");
        }

        @Mixin
        ExampleMixin mixin;

        @Override
        public Void call() throws Exception {
            return null;
        }
    }

    public static void main(String[] args) {
        CommandLine.call(new ExampleCommand(), args);
    }
```

When run with `--raptor off` (an option in the root `Callable`):
```
Didn't say the magic word.
```

When run with `--trex off` (an option in the mixin):
```
Could not invoke void com.zegelin.SpecRepro$ExampleMixin.setTRexFences(java.lang.String) with off
```
The `InvocationTargetException` cause is a NPE, from the assert @ `CommandLine.java:8852`, since the spec commandline is null.

_Version: 3.5.0_

**Comment by remkop** [comment]
Fixed in master.
I will do a release for this now.

**Comment by remkop** [comment]
The fix is included in piococli-3.5.1. Enjoy!
